src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,68 @@
 package core.basesyntax.service;
 
+import core.basesyntax.RegistrationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final String INVALID_PASSWORD_MESSAGE = "The provided password is invalid. "
+            + "It should include at least 6 characters!";
+    private static final String INVALID_LOGIN_MESSAGE = "The provided login is invalid. "
+            + "It should include at least 6 characters!";
+    private static final String LOGIN_EXISTS_MESSAGE = "The user with this login already exists.";
+    private static final String USER_IS_NULL_MESSAGE = "User cannot be null!";
+    private static final String AGE_INVALID_MESSAGE = "The age should be over 18.";
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
+    private StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        if (checkIfUserIsNotNull(user)
+                && validateAge(user.getAge())
+                && validateLogin(user.getLogin())
+                && validatePassword(user.getPassword())
+                && checkIfLoginExists(user.getLogin())) {
```

Comment:
I believe we do not need our validation methods to have boolean return type, as it causes redundant if statement. Feel free to make them void, anyway if validation fails - there will be an exception